### PR TITLE
HBCA-2 Search Service

### PR DESCRIPTION
**Pull request by heathharrelson**
Work on tests for the search component is taking longer than I'd like, and this has gone too long without a review. Here's the work I've done so far to create a wrapper that queries the Solr index.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -26,6 +26,8 @@ module.exports = {
     // allow debugger during development
     'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
     // require semicolons
-    'semi': ['error', 'always']
+    'semi': ['error', 'always'],
+    // don't require space before function parens
+    'space-before-function-paren': 'off'
```
Comment (heathharrelson) [inline .eslintrc.js]: I'm really not a fan of some of the style choices suggested by Vue.

File: src/services/search-service.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  /**
+   * Constructs the URL to query the Solr index for the given terms.
+   *
+   * @param {String} queryText - search terms to query. Required.
+   * @return {String} URL with query string to use to query the index.
+   * @private
+   */
+  _buildUrl(queryText) {
```
[the remark this answers is not in the document]
Comment (heathharrelson) [reply src/services/search-service.js]: The `defaultParams` are also kind of arbitrary. They'll probably need to be refined once I get this merged.

File: src/services/search-service.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    return `${this.searchUrl}${queryString}`;
+  }
+
+  /**
+   * Queries the Solr index with the given search terms.
+   *
+   * @param {String} queryText - search terms to query for.
+   * @return {Promise -> Object} a promise. On success, resolves to data returned by
+   *   the Solr query. On error, resolves an object with an error key containing the
+   *   reason the request rejected.
```
Comment (wsams) [inline src/services/search-service.js]: Minor text quibble - I think it should be `was rejected`.

File: .eslintrc.js (hunk 1 above)
Comment (wsams) [reply .eslintrc.js]: That would bother me too.

**Pull request merged**